cli/src/commands/install.js

```diff
@@ -63,19 +62,114 @@ export async function run(args: Args): Promise<number> {
   }
   const flowVersion = stringToVersion(flowVersionStr);
 
-  const term = args._[1];
+  // Find the project root
+  const projectRoot = await findFlowProjectRoot();
+  if (projectRoot === null) {
+    return failWithMessage(
+      `\nERROR: Unable to find a flow project in the currend dir or any of ` +
+      `it's parents!\nPlease run this command from within a Flow project.`
+    );
+  }
 
-  const matches = term.match(/([^@]*)@?(.*)?/);
-  const defName = (matches && matches[1]);
-  const defVersion = (matches && matches[2]) || 'auto';
+  // generate a map { dependency: version } for libs to install
+  // from command line or package.json.
+  let depsMap: DepsMap = {};  
+  if(args._[1]) {
+    const term = args._[1];
+    const matches = term.match(/([^@]*)@?(.*)?/);
+    if(!matches) {
+      console.error(      
+        "Please specify a package name of the format `PackageFoo` or " +
+        "PackageFoo@0.2.2"
+      );
+    }
+    
+    const defName = (matches && matches[1]);
+    const defVersion = (matches && matches[2]) || 'auto';
+    if(defName) {
+      depsMap[defName] = defVersion;
+    }
+  } else {
+    depsMap = await getInstalledPackageDependencies(process.cwd());
+  }
 
-  if (!defName) {
+  if (!Object.keys(depsMap).length === 0) {
     return failWithMessage(
-      "Please specify a package name of the format `PackageFoo` or " +
-      "PackageFoo@0.2.2"
+      "No package dependencies were found in package.json or specified on the command line."
     );
   }
 
+  // Get a list of defs to install
+  const defs: [LibDef] = [];
+  for(let dep in depsMap) {
+    console.log(`Found npm dependency: ${dep} v${depsMap[dep]}`);
+    let def = await findLibDef(
```

Comment:
You're probably not done here, but when you are: Best to group these `await`s into an array and use `Promise.all()` to dispatch them all together (rather than serially)